apps/prairielearn/src/admin_queries/generate_and_enroll_users.ts

```diff
@@ -44,7 +45,11 @@ export default async function ({
   let course_instance: CourseInstance | null = null;
   let course: Course | null = null;
   if (course_instance_id !== '') {
-    course_instance = await selectOptionalCourseInstanceById(course_instance_id);
+    course_instance = await selectOptionalCourseInstanceById({
+      id: course_instance_id,
+      requestedRole: 'System',
+      authzData: dangerousFullSystemAuthz(),
```

Comment:
Discussed with @reteps on Slack: this (and the necessary types in each model function that's auth-aware) look kind of gross, though they are very explicit.